test/e2e/gateway_api_test.go

```diff
@@ -360,3 +366,23 @@ func ensureGatewayObjectSuccess(t *testing.T, ns *corev1.Namespace) []string {
 
 	return errs
 }
+
+// createExperimentalCRDs creates experimental Gateway API custom resource definitions.
+func createExperimentalCRDs(t *testing.T) {
+	t.Helper()
```

Comment:
Is `createExperimentalCRDs` really a helper?  It looks more like a test.  If it failed, I'd want to get a line number of the point of failure within `createExperimentalCRDs`, not the call site of `createExperimentalCRDs`.  
```suggestion
// testExperimentalCRDs creates experimental Gateway API custom resource definitions.
func testExperimentalCRDs(t *testing.T) {
```

Reply:
I tried to stay in "the traditions" of this test file which uses `t.Helper()` for all the tests: [1](https://github.com/openshift/cluster-ingress-operator/blob/a41be6eed16361903e5b8ddd0131ac7585e87da3/test/e2e/gateway_api_test.go#L106-L107), [2](https://github.com/openshift/cluster-ingress-operator/blob/a41be6eed16361903e5b8ddd0131ac7585e87da3/test/e2e/gateway_api_test.go#L125-L126), etc. 

I think the wording `create` does sound like a helper indeed. While `test` doesn't quite reflect the logic of the function. I think `ensureExperimentalCRDs` can be the middle ground here, highlighting the testing nature of the function.

Reply:
> I tried to stay in "the traditions" of this test file which uses `t.Helper()` for all the tests: [1](https://github.com/openshift/cluster-ingress-operator/blob/a41be6eed16361903e5b8ddd0131ac7585e87da3/test/e2e/gateway_api_test.go#L106-L107), [2](https://github.com/openshift/cluster-ingress-operator/blob/a41be6eed16361903e5b8ddd0131ac7585e87da3/test/e2e/gateway_api_test.go#L125-L126), etc.

Yeah, I noticed those "traditions" and fixed them here: https://github.com/openshift/cluster-ingress-operator/pull/1152/commits/748bf50e7688f64a5ce5c2692553bf1835a1de73

The reason I don't want them to have `t.Helper()` is that it makes diagnosing failures more difficult.  While working on #1152, I would get failures with line numbers pointing me to the helper's call site, and I needed to search for the failure message or trace through the code to find the interesting part of the test.

Reply:
Removed `t.Helper()` from subtests.